src/uthash.h

```diff
@@ -26,9 +26,11 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #define UTHASH_VERSION 2.0.1
 
+#ifndef uthash_noheaders
```

Comment:
I'll rename it to free-standing, because that's what it's supposed to be. However, I will still snip out `<stdint.h>` (I'll have to check what out of stdint does uthash need though), as this can be documented and developer can include it if they need to. As my understanding goes so far, the world of embedded things don't follow the standards all too well. For example, Micirum has functions like Str_Len() and such, I'm not sure is it because they don't trust the compiler builtins or what not. At the same time, their headers override standard types left and right, so including things like `<stddef.h>` just generates a slough of warnings, and at worse may result into compiler referencing something from builtin library that has ends in libc, which in turn fails linking.